ideas/xx-message-ordering.md

```diff
@@ -0,0 +1,42 @@
+## Preamble
+
+    Idea: #??-message-ordering
+    Title: Message ordering
+    Status: Draft
+    Created: 2018-03-29
+
+
+## Summary
+
+Addresses point 2 and 3 of https://github.com/status-im/ideas/issues/99 .
+
+Define, document and implement message ordering for 1-to-1, group and public chats so that we can give the best UX experience to the end user.
+
+## Swarm Participants
+
+- Lead & PM: @cammellos
+- Testing & Evaluation: 
+- UX/Design: 
+
+## Product Overview
+
+Because status is a fully decentralized app, message ordering is non-trivial and needs careful thinking in terms of UX, as some constrainsts needs to be taken into consideration.
+
+Currently message ordering in the chat is by the time the message has been received which results in out-of-order messages and potentially different order on different devices.
+
+The primary aim of this swarm is to define which message ordering gives users the best UX experience and implement the changes necessary (if any).
+
+**Goals**
+- Define, document and implement ordering of messages for 1-to-1/group/public chats
+
+### Exit criteria
+
+- Message ordering is clearly defined and documented, including the rationale behind the choices made
+- Code reflecting the choice made is written and released
+- Message ordering implementation is covered by integration tests (at least 80% coverage, as per OKR)
+
+### Success metrics
+
+- A/B testing of different implementations clearly shows a preference for the chosen ordering
+- Ordering of messages can be understood by the end-user through documentation (survey)
```

Comment:
What does it mean for end-user to understand it through documentation?

Reply:
It's a bit difficult to express, but the idea is that he understands the limitation of ordering in decentralized systems.

Having an 100% accurate system is impossible as we know, so it's important for the user to understand its limit, to avoid disappointment.

Say we decide to go for received time (just as an example), then it's important to communicate to the user that some messages might be received out of order. 

The idea would be to have some docs that clearly explain that to the user. 
Probably should be phrased better I take :)

Reply:
Gotcha, so maybe part of onboarding/education then? cc @andmironov as this has been talked about a bunch  - probably worth seeing his POV

Reply:
precisely! I will look at it.